design-docs/improved-console.md

```diff
@@ -0,0 +1,283 @@
+# Improved Console Output
+
+## Primary Goals
+* Improve perceived performance by making user aware of background work
+* Modern feel
+* User extensibility
+
+### User Goals for Console
+In priority order, the user wants to answer the following questions:
+
+* When will my build finish?
+    * Soon (should I keep watching it) or later ("I'll check back")
+* Is there anything I should know about my build right now? (e.g. tests have failed, warnings)
+* What is Gradle doing right now?
+* Is my build successful?
+* What command did I execute?
+
+## Story: Display parallel work in-progress independently by operation
```

Comment:
How will the functionality be introduce to the user? Is it an opt-in feature? Can be turned off/on.